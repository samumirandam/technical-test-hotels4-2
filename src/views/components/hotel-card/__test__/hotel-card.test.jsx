/* eslint-disable import/no-unresolved */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render, screen, fireEvent } from '@utils/test-utils';

import HotelCard from '../index';

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<HotelCard {...setupStore} />, {});
};

describe('Test for HotelCard component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByText('Secret price available')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('Should render click in secret price', () => {
    setup();
    fireEvent.click(screen.getByText('Secret price available'));
    expect(screen.getByText('Secret price available')).toBeTruthy();
  });
});
