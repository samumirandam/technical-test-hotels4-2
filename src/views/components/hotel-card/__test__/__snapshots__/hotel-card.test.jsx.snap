// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for HotelCard component Should renders the same component 1`] = `
<div>
  <div
    class="HotelCard"
  >
    <figure>
      <img
        alt=""
        src=""
      />
    </figure>
    <div
      class="HotelCard__content"
    >
      <div>
        <p
          class="HotelCard__title"
        />
        <div
          class="HotelCard__price-container"
        >
          <p
            class="HotelCard__price"
          />
          <p
            class="HotelCard__price-info"
          />
        </div>
        <p
          class="HotelCard__total-price"
        />
        <div
          class="HotelCard__raiting-container"
        >
          <p
            class="HotelCard__raiting"
          />
          <p
            class="HotelCard__opinions"
          />
        </div>
        <div
          class="HotelCard__badges"
        />
      </div>
      <button
        class="Button HotelCard__secret-price"
        type="button"
      >
        Secret price available
      </button>
    </div>
  </div>
</div>
`;
